src/pages/docs/chat/rooms/media-sharing.mdx

```diff
@@ -0,0 +1,165 @@
+Let's say that the images will be served from `https://example.com/images/<identifier>`, where the identifier is a string of length between 10 and 15 and matches the `^[a-z0-9]{10,15}$` regex.
+
+### Sending messages with images
+
+When the user initiates the upload flow (for example by clicking a button called "attach"), we use the `uploadImage()` flow and save the resulting ID, like this:
+
+```javascript
+let imagesToAttach = [];
+async function onImageAttach() {
+  const imageId = await uploadImage();
+  imagesToAttach.push(imageId);
+}
+```
+
+In the UI, the `imagesToAttach` array is displayed so the users know what images will be attached to their message. The UI can allow users to remove or sort the images.
+
+When a user sends a message we check if there are any images to attach, and then send the message with appropriate metadata:
+
+```javascript
+async function send(text) {
+  let metadata = {};
+  if (imagesToAttach.length > 0) {
+    metadata["images"] = imagesToAttach;
+    imagesToAttach = [];
+  }
+  await room.messages.send({
+    text: text,
+    metadata: metadata
+  });
+}
+```
+
+### Displaying images
+
+First we need a way to get the valid image IDs, if any, from a message:
+
+```javascript
+const imageIdRegex = /^[a-z0-9]{10,15}$/;
+
+function getValidImages(message) {
+  if (message.metadata.images && message.metadata.images.length > 0) {
+    return message.metadata.images.filter(imageId => imageIdRegex.test(imageId));
+  }
+  return [];
+}
+```
+
+And now for each message we can display images if it has any.
+We provide here a simplistic example for displaying text and images by constructing the DOM for them manually. The same high-level logic applies for modern JavaScript front-end frameworks, Swift, and Kotlin.
+
+```javascript
+function createMessageDOM(message) {
+  const container = document.createElement("div");
+  const text = document.createElement("div");
+  text.innerText = message.text;
+  container.appendChild(text);
+  const validImages = getValidImages(message);
+  if (validImages.length > 0) {
+    const imagesContainer = document.createElement("div");
+    for (let imageId of validImages) {
+      const img = document.createElement("img");
+      img.src = "https://example.com/images/"+imageId;
+      imagesContainer.appendChild(img);
+    }
+    container.appendChild(imagesContainer);
+  }
+  return container;
+}
+```
+
+### Why image ID and not URL?
+
+Image IDs are simpler to validate than full URLs. You can also use URLs, but you must correctly validate them on the receiving end.
+
+### Metadata for each image
+
+In the example above we used a simple string array to store image IDs. We can change this to be an object where each image holds extra information that can be displayed to the user, such as a title or description, or the expected size in pixels so that the UI can allocate a placeholder loading box of the correct proportions.
```

Comment:
Should this be titled differently and provide an example?

Reply:
I've removed this and instead made it the default.